openmls/src/treesync/mod.rs

```diff
@@ -79,24 +81,77 @@ pub use node::{leaf_node::LeafNode, parent_node::ParentNode, Node};
 pub mod tests_and_kats;
 
 /// An exported ratchet tree as used in, e.g., [`GroupInfo`](crate::messages::group_info::GroupInfo).
-#[derive(
-    PartialEq,
-    Eq,
-    Clone,
-    Debug,
-    Default,
-    Serialize,
-    Deserialize,
-    TlsSerialize,
-    TlsDeserialize,
-    TlsSize,
-)]
-// TODO(#1305): Make the ratchet tree non-malleable.
-//
-// * ValSemXXX: "If the tree has 2^d leaves, then it has 2^(d+1) - 1 nodes. The ratchet_tree vector logically has this number of entries, but the sender SHOULD NOT include blank nodes after the last non-blank node."
-// * ValSemXXX: "Regardless of how the client obtains the tree, the client MUST verify that the root hash of the ratchet tree matches the tree_hash of the GroupContext before using the tree for MLS operations."
```

Comment:
I removed the second ValSem for now but we should get to this during valiadtion.